pkg/kritis/cryptolib/pgp.go

```diff
@@ -51,15 +52,72 @@ func (v pgpVerifierImpl) verifyPgp(signature, publicKey []byte) ([]byte, error)
 	// This will call PublicKey.VerifySignature for the keys in the keyring.
 	payload, err := ioutil.ReadAll(messageDetails.UnverifiedBody)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not verify armor signature")
+		return nil, errors.Wrap(err, "error reading message contents")
 	}
 
-	// Make sure after reading the UnverifiedBody above, there is no signature error.
+	// Make sure after reading the UnverifiedBody above that the Signature
+	// exists and there is no SignatureError.
 	if messageDetails.SignatureError != nil {
-		return nil, errors.Wrap(messageDetails.SignatureError, "bad signature found")
+		return nil, errors.Wrap(messageDetails.SignatureError, "failed to validate: signature error")
 	}
 	if messageDetails.Signature == nil {
-		return nil, errors.New("no signature found for given key")
+		return nil, fmt.Errorf("failed to validate: signature missing")
 	}
 	return payload, nil
 }
+
+type pgpSigner struct {
+	privateEntity *openpgp.Entity
+	publicKeyID   string
+}
+
+// NewPgpSigner creates a Signer interface for PGP Attestations. `privateKey`
+// contains the ASCII-armored private key.
+func NewPgpSigner(privateKey []byte) (Signer, error) {
```

Comment:
Should this be string instead of []byte?

Reply:
Acknowledged, choosing to keep this as is since Go crypto libraries represent key material as `[]byte`.